File: src/main/java/net/ndrei/teslacorelib/inventory/SidedItemHandlerWrapper.java
package net.ndrei.teslacorelib.inventory;

import net.minecraft.item.ItemStack;
import net.minecraft.util.EnumFacing;
import net.minecraftforge.items.IItemHandler;

import javax.annotation.Nonnull;

/**
 * Created by CF on 2016-12-17.
 */
public class SidedItemHandlerWrapper implements IItemHandler {
    private int[] slots;
    private SidedItemHandler handler;
    private EnumFacing side;

    public SidedItemHandlerWrapper(SidedItemHandler handler, EnumFacing side) {
        this.slots = (this.handler = handler).getSlotsForFace(this.side = side);
    }

    @Override
    public int getSlots() {
        return this.slots.length;
    }

    @Nonnull
    @Override
    public ItemStack getStackInSlot(int slot) {
        return this.handler.getStackInSlot(this.slots[slot]);
    }

    @Nonnull
    @Override
    public ItemStack insertItem(int slot, @Nonnull ItemStack stack, boolean simulate) {
        return this.handler.insertItem(this.slots[slot], stack, simulate);
    }

    @Nonnull
    @Override
    public ItemStack extractItem(int slot, int amount, boolean simulate) {
        return this.handler.extractItem(this.slots[slot], amount, simulate);
    }

//    @Override
//    public int getSlotLimit(int slot) {
//        return this.handler.getSlotLimit(this.slots[slot]);
//    }
}

File: src/main/kotlin/net/ndrei/teslacorelib/inventory/SidedItemHandlerWrapper.kt
package net.ndrei.teslacorelib.inventory

import net.minecraft.item.ItemStack
import net.minecraft.util.EnumFacing
import net.minecraftforge.items.IItemHandler

/**
 * Created by CF on 2017-06-28.
 */
class SidedItemHandlerWrapper(private val handler: SidedItemHandler, private val side: EnumFacing) : IItemHandler {
    private val slots: IntArray
            = this.handler.getSlotsForFace(this.side)

    override fun getSlots()
            = this.slots.size

    override fun getStackInSlot(slot: Int)
            = this.handler.getStackInSlot(this.slots[slot])

    override fun insertItem(slot: Int, stack: ItemStack, simulate: Boolean)
            = this.handler.insertItem(this.slots[slot], stack, simulate)

    override fun extractItem(slot: Int, amount: Int, simulate: Boolean)
            = this.handler!!.extractItem(this.slots[slot], amount, simulate)

    override fun getSlotLimit(slot: Int)
            = this.handler!!.getSlotLimit(this.slots[slot])
}
